link-grammar/prune.c

```diff
@@ -573,14 +573,14 @@ static inline bool matches_S(connector_table *ct, Connector * c, char dir)
  */
 static int mark_dead_connectors(connector_table *ct, Exp * e, char dir)
 {
-	int count;
-	count = 0;
+	int count = 0;
+	Connector dummy;
+	init_connector(&dummy);
```

Comment:
You are right. This was outlandish, including the comment (no loop there). It even increases the overhead, as it performes the initialization even when it has no use.
The only  way to eliminate this initialization is to restructure the code to use the connector directly.

I tried to touch this code because mark_dead_connectors() is in the top-10 hot functions.

So I can resend this pull request without this commit.